starfish/ddo/authentication.py

```diff
@@ -0,0 +1,91 @@
+"""
+    Authentication Class
+    To handle embedded public keys
+"""
+
+import json
+import re
+
+from squid_py.ddo.public_key_base import PublicKeyBase
+
+
+class Authentication:
+    """DDO Authentication"""
+
+    def __init__(self, key_id, authentication_type):
+        """ init an authentication based on it's key_id and authentication type"""
```

Comment:
could you describe what exactly is/are the authentication_type(s) in the docstring?

Reply:
This is the missing class taken out of squid-py, so need to add it back in so that we can provide different authentications for DDO's